lib/batch/batch_metadata_updates_654.rb

```diff
@@ -0,0 +1,53 @@
+# batch updates for https://github.com/galterlibrary/digital-repository/issues/654
+
+def add_digitalhub_to_non_private_empty_publisher_record 
+  empty_publisher_items = GenericFile.where(publisher: [])+ 
+                          Collection.where(publisher: [])
+  puts "updating #{empty_publisher_items.count} records with empty publisher"
+  empty_publisher_items.each do |item|
+    if item.visibility != "restricted"
+      item.publisher += ["DigitalHub. Galter Health Sciences Library & Learning Center"]
+      item.save!
+    end
+  end 
+end
+
+def add_digitalhub_to_publisher(publisher)
+  solr_items = ActiveFedora::SolrService.query(
+    # _sim returns an exact match
+    "publisher_sim:\"#{publisher}\"",
+    { rows: 99999 }
+  )
+  puts "updating ALL(#{solr_items.count}) records with publisher #{publisher}"
+  solr_items.each do |solr_item|
+    constant_item = solr_item['has_model_ssim'].first.constantize
+    item = constant_item.find(solr_item['id'])
+
+    # convert to array for easier manipulation
+    publisher_array = item.publisher.to_a
+    publisher_array.map! { |pub|
```

Comment:
huh interesting! I probably would have gone for the single replacement (`index(publisher)` route) but that works.

Reply:
I will keep it as is, but is `.index(publisher)` more preferable? I originally did try it this way. Settled on `.map!` for no specific reason, but it would catch any items with duplicate `publisher`, if that matters.

Reply:
The `.index` is probably faster because not *every* publisher in the list will be processed. But optimization here doesn't really matter if the whole operation went fast enough. I had seen how `.map!` would deal with duplicate publishers, but duplicate publishers should be an incorrect data state anyway. This is fine.